app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -0,0 +1,312 @@
+    _log.warn($ERROR_INFO)
+    raise MiqException::MiqUnreachableError, $ERROR_INFO
+  rescue Ovirt::MissingResourceError, URI::InvalidURIError
+    raise MiqException::MiqUnreachableError, "Invalid URI specified for the server."
+  rescue RestClient::Unauthorized
+    raise MiqException::MiqInvalidCredentialsError, "Incorrect user name or password."
+  rescue
+    _log.error("Error while verifying credentials #{$ERROR_INFO}")
+    raise MiqException::MiqEVMLoginError, $ERROR_INFO
+  end
+
+  def rhevm_metrics_connect_options(options = {})
+    metrics_hostname = connection_configuration_by_role('metrics')
+      .try(:endpoint)
+      .try(:hostname)
+    server   = options[:hostname] || metrics_hostname || hostname
+    username = options[:user] || authentication_userid(:metrics)
+    password = options[:pass] || authentication_password(:metrics)
+    database = options[:database]
+
+    {
+      :host     => server,
+      :database => database,
+      :username => username,
+      :password => password
+    }
+  end
+
+  def verify_credentials_for_rhevm_metrics(options = {})
+    require 'ovirt_metrics'
+    OvirtMetrics.connect(rhevm_metrics_connect_options(options))
+    OvirtMetrics.connected?
+  rescue PGError => e
+    message = (e.message.starts_with?("FATAL:") ? e.message[6..-1] : e.message).strip
+
+    case message
+    when /database \".*\" does not exist/
+      if database.nil? && (conn_info[:database] != OvirtMetrics::DEFAULT_HISTORY_DATABASE_NAME_3_0)
+        conn_info[:database] = OvirtMetrics::DEFAULT_HISTORY_DATABASE_NAME_3_0
+        retry
+      end
+    end
+
+    _log.warn("PGError: #{message}")
+    raise MiqException::MiqEVMLoginError, message
+  rescue Exception => e
+    raise MiqException::MiqEVMLoginError, e.to_s
+  ensure
+    OvirtMetrics.disconnect rescue nil
+  end
+
+  def authentications_to_validate
+    at = [:default]
+    at << :metrics if self.has_authentication_type?(:metrics)
+    at
+  end
+
+  def verify_credentials(auth_type = nil, options = {})
+    auth_type ||= 'default'
+
+    case auth_type.to_s
+    when 'default' then verify_credentials_for_rhevm(options)
+    when 'metrics' then verify_credentials_for_rhevm_metrics(options)
+    else;          raise "Invalid Authentication Type: #{auth_type.inspect}"
+    end
+  end
+
+  def history_database_name
+    @history_database_name ||= begin
+                                 version = version_3_0? ? '3_0' : '>3_0'
+                                 self.class.history_database_name_for(version)
+                               end
+  end
+
+  def version_3_0?
+    if @version_3_0.nil?
+      @version_3_0 =
+        if api_version.nil?
+          with_provider_connection(&:version_3_0?)
+        else
+          api_version.starts_with?("3.0")
+        end
+    end
+
+    @version_3_0
+  end
+
+  class_methods do
+
+    def api3_supported_features
+      []
+    end
+
+    def api4_supported_features 
+      [:reconfigure_vms, :snapshots]
```

Comment:
It is https://github.com/ManageIQ/manageiq/pull/11418/files#diff-0613648e8f7d7c13b5aeb5071c90f6d2R73
and the validate exists for the snapshot, and to be added after this patch for reconfigure_vms.